### ISSUE TITLE
Alert close button names are too generic

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3498]
## Describe the bug

The close button on an alert can be too generic in cases where there are multiple close buttons on the page (like inside a modal or with more than one alert open), and screen readers should have better details read out.

## How to reproduce

https://stackblitz.com/github/clane/clarityTest2/tree/master/clarity-test-app

- Using a screen reader, activate the WIZARD + ASYNC FORM VALIDATION button under the Wizards - Wizard with Async Form Validation example
- Activate the next button
- Locate and activate the Close button associated with the alert that contains the message "Your answer is incorrect"

## Expected behavior

Provide a mechanism for the developer using clarity to pass in a specific accessible name for the close button for any alert. Simply using the text 'close' is not specific enough since there may be similar buttons on the page for other controls. Close can be the fallback, but expect the applications to provide meaningful close text.

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

There are close buttons in many places, so this basic solution may need to be applied elsewhere.

**Comment by hippee-lee** [comment]
Provide an input/property for consumer to describe the close button for the alert. 

Nice to have, log a warning if/when this Input/property isn't set in dev mode.
